File: js/5.js
/* 
* @Author: ben_cripps
* @Date:   2014-12-05 20:04:00
* @Last Modified by:   ben_cripps
* @Last Modified time: 2014-12-05 20:17:27
*/

'use strict';

function isDivisibleBy(f, l, n) {
    for ( var i = f; i <= l; i++ ) {

        if ( n % i !== 0 ) return false;

    }
    return true;
}   

function getSmallest(n) {
    var fin = false;
    while (!fin) {
        if ( !isDivisibleBy(1,20, n) ){
            isDivisibleBy(1,20, n+=1);
        }
        else {
            fin = true;
        }
    }
    return n;
}

console.log( getSmallest(1) );

File: python/5.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# @Author: ben_cripps
# @Date:   2015-01-03 10:33:19
# @Last Modified by:   ben_cripps
# @Last Modified time: 2015-01-03 14:10:27
import math

def isPrime(n):
    return n != 1 and len([x for x in range(2, n) if n % x == 0]) == 0

def get_exponent(n,x):
    return math.floor(math.log(n,x))

def smallest_number_divisible_by(n):
    prime_factors = [x for x in range(2,n) if isPrime(x)]
    number_of_primes = map(lambda x: math.pow(x, get_exponent(n,x)), prime_factors)
    return reduce(lambda x,y: x*y, number_of_primes)

print(smallest_number_divisible_by(20))